jenkins/opensearch/benchmark-pull-request.jenkinsfile

```diff
@@ -0,0 +1,311 @@
+                    [key: 'INGEST_NODE_COUNT', value: '$.INGEST_NODE_COUNT'],
+                    [key: 'ML_NODE_COUNT', value: '$.ML_NODE_COUNT'],
+                    [key: 'DATA_INSTANCE_TYPE', value: '$.DATA_INSTANCE_TYPE'],
+                    [key: 'DATA_NODE_STORAGE', value: '$.DATA_NODE_STORAGE'],
+                    [key: 'ML_NODE_STORAGE', value: '$.ML_NODE_STORAGE'],
+                    [key: 'JVM_SYS_PROPS', value: '$.JVM_SYS_PROPS'],
+                    [key: 'ADDITIONAL_CONFIG', value: '$.ADDITIONAL_CONFIG'],
+                    [key: 'USE_50_PERCENT_HEAP', value: '$.USE_50_PERCENT_HEAP'],
+                    [key: 'USER_TAGS', value: '$.USER_TAGS'],
+                    [key: 'WORKLOAD_PARAMS', value: '$.WORKLOAD_PARAMS'],
+                    [key: 'TEST_PROCEDURE', value: '$.TEST_PROCEDURE'],
+                    [key: 'EXCLUDE_TASKS', value: '$.EXCLUDE_TASKS'],
+                    [key: 'INCLUDE_TASKS', value: '$.INCLUDE_TASKS'],
+                    [key: 'CAPTURE_NODE_STAT', value: '$.CAPTURE_NODE_STAT'],
+                    [key: 'TELEMETRY_PARAMS', value: '$.TELEMETRY_PARAMS']
+                ],
+                tokenCredentialId: 'jenkins-pr-benchmark-generic-webhook-token',
+                causeString: 'Triggered by comment on PR on OpenSearch core repository',
+                printContributedVariables: true,
+                printPostContent: true
+            )
+    }
+
+    stages {
+        stage('validate-and-set-parameters') {
+            agent { label AGENT_LABEL }
+            steps {
+                script {
+                    if (DISTRIBUTION_URL == '' || DISTRIBUTION_VERSION == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Benchmark Tests failed to start. Provide DISTRIBUTION_URL and DISTRIBUTION_VERSION to run tests")
+                    }
+                    env.ARCHITECTURE = "x64"
+                    lib.jenkins.Messages.new(this).add(JOB_NAME, "Benchmark tests for ${DISTRIBUTION_URL}")
+                    if (currentBuild.rawBuild.getCauses().toString().contains("GenericCause")) {
+                        currentBuild.description = "Benchmark initiated by PR:${pull_request} on ${repository}"
+                    }
+                    else {
+                        currentBuild.description = "Running benchmark test for distribution-url: ${DISTRIBUTION_URL} distribution-version: ${DISTRIBUTION_VERSION}"
+                    }
+                }
+            }
+            post {
+                always {
+                    postCleanup()
+                }
+            }
+        }
+        stage('benchmark-pull-request') {
+            agent { label AGENT_LABEL }
+            steps {
+                script {
+                    echo "security-enabled: ${SECURITY_ENABLED}"
+
+                    runBenchmarkTestScript(
+                        distributionUrl: DISTRIBUTION_URL,
```

Comment:
Should all of these values have `$` or `params`? Just confirming

Reply:
Not required.